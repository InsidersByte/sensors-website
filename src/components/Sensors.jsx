/* @flow */

import React from 'react';
import styled from 'styled-components';
import Sensor from './Sensor';
import type { SensorType } from '../types';

const Wrapper = styled.div`
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
`;

const Sensors = ({ sensors }: { sensors: Array<SensorType> }) => (
    <Wrapper>
        {sensors.map(sensor => (
            <Sensor key={sensor.id} sensor={sensor} />
        ))}
    </Wrapper>
);

export default Sensors;
