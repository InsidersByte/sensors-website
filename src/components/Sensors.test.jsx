import React from 'react';
import { shallow } from 'enzyme';
import Sensors from './Sensors';

describe('Sensors', () => {
    it('should render', () => {
        const wrapper = shallow(
            <Sensors sensors={[
                { id: 1 },
                { id: 2 },
            ]} />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
