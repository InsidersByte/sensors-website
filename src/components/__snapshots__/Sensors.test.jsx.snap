exports[`Sensors should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <Sensor
        sensor={
            Object {
                "id": 1,
              }
        } />
    <Sensor
        sensor={
            Object {
                "id": 2,
              }
        } />
</styled.div>,
  "nodes": Array [
    <styled.div>
      <Sensor
            sensor={
                  Object {
                        "id": 1,
                      }
            } />
      <Sensor
            sensor={
                  Object {
                        "id": 2,
                      }
            } />
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Sensors
        sensors={
                Array [
                        Object {
                          "id": 1,
                        },
                        Object {
                          "id": 2,
                        },
                      ]
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "sensors": Array [
            Object {
              "id": 1,
            },
            Object {
              "id": 2,
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <Sensor
                    sensor={
                              Object {
                                        "id": 1,
                                      }
                    } />
          <Sensor
                    sensor={
                              Object {
                                        "id": 2,
                                      }
                    } />
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <Sensor
                    sensor={
                              Object {
                                        "id": 1,
                                      }
                    } />
          <Sensor
                    sensor={
                              Object {
                                        "id": 2,
                                      }
                    } />
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Sensors
    sensors={
        Array [
            Object {
              "id": 1,
            },
            Object {
              "id": 2,
            },
          ]
    } />,
}
`;
